src/actuator/ActuatorBulkFAST.C

```diff
@@ -0,0 +1,308 @@
+            break;
+          }
+
+          for (int i = 0; i < 3; ++i) {
+            ThrowAssertMsg(
+              epsilonLocal(i) > 0.0,
+              "Epsilon zero for point: " + std::to_string(np) + " index " +
+                std::to_string(i));
+          }
+
+          // The radius of the searching. This is given in terms of
+          //   the maximum of epsilon.x/y/z/.
+          searchRadius_.h_view(np + offset) =
+            std::max(
+              epsilonLocal(0), std::max(epsilonLocal(1), epsilonLocal(2))) *
+            sqrt(log(1.0 / 0.001));
+        }
+      } else {
+        NaluEnv::self().naluOutput()
+          << "Proc " << NaluEnv::self().parallel_rank() << " glob iTurb "
+          << iTurb << std::endl;
+      }
+    }
+  }
+  actuator_utils::reduce_view_on_host(epsilon_.view_host());
+  actuator_utils::reduce_view_on_host(epsilonOpt_.view_host());
+  actuator_utils::reduce_view_on_host(searchRadius_.view_host());
+  epsilon_.sync_host();
+  epsilonOpt_.sync_host();
+  searchRadius_.sync_host();
+}
+
+Kokkos::RangePolicy<ActuatorFixedExecutionSpace>
+ActuatorBulkFAST::local_range_policy()
+{
+  auto rank = NaluEnv::self().parallel_rank();
+  if (rank == openFast_.get_procNo(rank)) {
+    const int offset = turbIdOffset_.h_view(rank);
+    const int size = openFast_.get_numForcePts(rank);
+    return Kokkos::RangePolicy<ActuatorFixedExecutionSpace>(
+      offset, offset + size);
+  } else {
+    return Kokkos::RangePolicy<ActuatorFixedExecutionSpace>(0, 0);
+  }
+}
+
+void
+ActuatorBulkFAST::interpolate_velocities_to_fast()
+{
+  if (!openFast_.isDryRun()) {
+    openFast_.interpolateVel_ForceToVelNodes();
+
+    if (openFast_.isTimeZero()) {
+      openFast_.solution0();
+    }
+  }
+}
+
+void
+ActuatorBulkFAST::step_fast()
+{
+  if (!openFast_.isDryRun()) {
+    for (int j = 0; j < tStepRatio_; j++) {
+      openFast_.step();
+    }
+  }
+}
+
+bool
+ActuatorBulkFAST::fast_is_time_zero()
+{
+  int localFastZero = (int)openFast_.isTimeZero();
+  int globalFastZero = 0;
+  MPI_Allreduce(
+    &localFastZero, &globalFastZero, 1, MPI_INT, MPI_SUM,
+    NaluEnv::self().parallel_comm());
+  return globalFastZero > 0;
+}
+
+void
+ActuatorBulkFAST::output_torque_info()
+{
+  for (size_t iTurb = 0; iTurb < turbineThrust_.extent(0); iTurb++) {
+
+    int processorId = openFast_.get_procNo(iTurb);
+
+    if (NaluEnv::self().parallel_rank() == processorId) {
+      auto thrust = Kokkos::subview(turbineThrust_, iTurb, Kokkos::ALL);
+      auto torque = Kokkos::subview(turbineTorque_, iTurb, Kokkos::ALL);
+      NaluEnv::self().naluOutput()
+        << std::endl
+        << "  Thrust[" << iTurb << "] = " << thrust(0) << " " << thrust(1)
+        << " " << thrust(2) << " " << std::endl;
+      NaluEnv::self().naluOutput()
+        << "  Torque[" << iTurb << "] = " << torque(0) << " " << torque(1)
+        << " " << torque(2) << " " << std::endl;
+
```

Comment:
Can we please implement some verbosity flags to control this output. It will be really useful for regression tests where this output is unnecessary can should be turned off.

Reply:
It is already there. This is only output when the turbine is out in debug mode. We just need to switch the regression tests to ‘debug: false’.  FWIW I do need to implement an output capture for OpenFAST though. It is spitting way to much stuff out in the unit tests now.